components/raftstore-v2/src/router/message.rs

```diff
@@ -207,64 +92,88 @@ impl StoreTick {
     }
 }
 
-/// Raft command is the command that is expected to be proposed by the
-/// leader of the target raft group.
-#[derive(Debug)]
-pub struct RaftCommand<S: Snapshot> {
+/// Command that can be handled by raftstore.
+pub struct RaftCommand {
     pub send_time: Instant,
     pub request: RaftCmdRequest,
-    pub ch: ResponseChannel<S>,
 }
 
-impl<S: Snapshot> RaftCommand<S> {
-    #[inline]
-    pub fn new(request: RaftCmdRequest, ch: ResponseChannel<S>) -> RaftCommand<S> {
+impl RaftCommand {
+    pub fn new(request: RaftCmdRequest) -> Self {
         RaftCommand {
+            send_time: Instant::now(),
             request,
+        }
+    }
+}
+
+pub struct ReadCommand {
```

Comment:
So the original code is to use the same RaftCommand for both read and write. 
Now the RaftCommand does not have the response channel, and instead response channel is move to the upper layer ReadCommand/WriteCommand. 

From the PR, I don't see there's need to differentiate Read/Write Command. What's the change in the future that would need to separate read and write command? To me the real source of truth is always in RaftCmdRequest. So what if WriteCommand has the read RaftCmdRequest?  I think the original code is fine.

Reply:
Read and write are always separated inside raftstore. If they are mixed in one command, the command will be rejected. It's the same as V1. Mixing read and write in the public interface adds more unnecessary encapsulation.